### PR TITLE
complete refactoring to respect streams and to avoid command-line argument

### PR DESCRIPTION
**Pull request by JohannesHoppe**
- no hard coded values (uses Config object)
- does not start a new child process
- TODO: tests! ;-)

fixes koemei/gulp-node-inspector#4
fixes koemei/gulp-node-inspector#5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,32 +13,43 @@ Perfect for development.
...
-    'save-live-edit': false,
-    preload: false,
-    'stack-trace-limit': 4
-  });
+var nodeInspector = require('gulp-node-inspector');
+
+gulp.task('debug', function() {
+
+  gulp.src(['app.js']) // any given file is ignored
+    .pipe(nodeInspector());
```
Comment (vohtaski) [inline README.md]: if the file `app.js` does not exist, it silently does not start.
Is it possible to do something like `gulp-karma` that I can run as 
```js
  return gulp.src([])
    .pipe(plugins.karma({}));
```

Comment (vohtaski) [inline README.md]: I guess the easiest is to move debug-server starting from `start` function,
they if you pass `src([])` it will start the node-inspector

Comment (JohannesHoppe) [inline README.md]: Yes, I see your point. I'm going to change it a little bit...

**Comment by JohannesHoppe** [comment]
All right, the third commit should do the trick. gulp-karma was a good example. I'm pretty satisfied with the result, but I'm still wondering how to trigger the 'close' event. Did I missed somewhere a button in the DevTools?

**Comment by JohannesHoppe** [comment]
see https://github.com/node-inspector/node-inspector/issues/547

**Comment by vohtaski** [comment]
lgtm, merging

**Pull request merged**